lib/libc/riscv/SYS.h

```diff
@@ -42,22 +42,46 @@
 	li	t0, SYS_ ## name;				\
 	ecall
 
+#ifndef __CHERI_PURE_CAPABILITY__
```

Comment:
Slight preference for using #ifdef and putting purecap case first as you did in <machine/asm.h>